### PR TITLE
feat(validation): add min/max deposit limits from PaymasterVault (#43)

### PR DESCRIPTION
**Pull request by rube-de**
## Summary

- Add input validation against PaymasterVault's `getTokenConfig` min/max deposit limits
- Disable bridge button with descriptive messages when amount is out of range
- Cap MAX button at `min(balance, maxDeposit)` to prevent limit violations
- Block bridge during limit loading to prevent premature gas-wasting submissions

## Changes

| File | Change |
|------|--------|
| `PaymasterVault_ABI.json` | Replace minimal ABI with full contract ABI |
| `paymasterVault.ts` | Add `getTokenConfig` imperative read function |
| `hooks/useDepositLimits.ts` | New hook: fetches/caches limits via `useReadContract` |
| `App.tsx` | Validation logic, button text, MAX button cap |

## Validation States

- **Below minimum** → Button: "Below minimum (10 USDC)", input turns red
- **Above maximum** → Button: "Above maximum (1,000 USDC)", input turns red
- **Token disabled** → Button: "Token not supported"
- **Contract error** → Button: "Unable to verify limits"
- **Loading** → Bridge blocked until limits verified

## Edge Cases

- Zero min/max → Treated as unconfigured (no validation)
- Contract read fails → Bridge blocked, shows error message
- Chain/token switch → Limits refetched automatically (TanStack Query)

## Test plan

- [ ] Select USDC on Base, enter amount below contract's minAmount → button shows "Below minimum"
- [ ] Enter amount above maxAmount → button shows "Above maximum"
- [ ] Enter amount within range → button shows "Bridge to Sapphire"
- [ ] Click MAX with balance > maxDeposit → amount capped at maxDeposit
- [ ] Switch token/chain → limits update without stale validation
- [ ] Disconnect wallet → no contract read errors in console

Fixes #43

### CODE REVIEW

File: frontend/src/hooks/useDepositLimits.ts [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    // Format with locale separators for display
+    const formatWithSeparators = (value: bigint): string => {
+      const formatted = formatUnits(value, decimals)
+      // Parse and format with commas, removing trailing decimals if whole number
+      const num = parseFloat(formatted)
+      return num.toLocaleString('en-US', {
+        maximumFractionDigits: decimals,
+        minimumFractionDigits: 0,
+      })
+    }
```
[the remark this answers is not in the document]
Comment (rube-de) [reply frontend/src/hooks/useDepositLimits.ts:82]: Valid catch — while our USDC/USDT limits won't realistically hit `Number.MAX_SAFE_INTEGER`, the BigInt-based approach is cleaner and eliminates the precision footgun entirely. Applied your suggested fix in 9c76dec.

File: frontend/src/contracts/paymasterVault.ts [hunk 2]
```diff
@@ -1,10 +1,38 @@
...
+    abi: PaymasterVault_ABI,
+    functionName: 'getTokenConfig',
+    args: [tokenAddress],
+    chainId,
+  })
+
+  const [minAmount, maxAmount, enabled] = result as [bigint, bigint, boolean]
+
+  return { minAmount, maxAmount, enabled }
+}
```
[the remark this answers is not in the document]
Comment (rube-de) [reply frontend/src/contracts/paymasterVault.ts]: Good point — `getTokenConfig` was from an earlier iteration before switching to `useReadContract` directly in the hook. Removed along with the unused `readContract` import in 9c76dec.

File: frontend/src/hooks/useDepositLimits.ts [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+      refetchOnWindowFocus: false,
+    },
+  })
+
+  return useMemo(() => {
+    if (isLoading || !data) {
+      return { ...DEFAULT_LIMITS, isLoading, isError }
+    }
+
+    // Result is tuple: [minAmount, maxAmount, enabled]
```
[the remark this answers is not in the document]
Comment (rube-de) [reply frontend/src/hooks/useDepositLimits.ts]: Agreed — the comment should match the local variable names since that's what the reader sees in the next line. Fixed in 9c76dec.

File: frontend/src/contracts/paymasterVault.ts [hunk 4]
```diff
@@ -1,10 +1,38 @@
...
 import PaymasterVault_ABI from './PaymasterVault_ABI.json'
 import { waitForTransactionReceipt } from '@wagmi/core'
 import { ROFL_PAYMASTER_DEPOSIT_GAS_LIMIT } from '../constants/rofl-paymaster-config.ts'
 import { config } from '../wagmi.ts'
 
+export interface TokenConfigResult {
+  minAmount: bigint
+  maxAmount: bigint
+  enabled: boolean
+}
```
[the remark this answers is not in the document]
Comment (rube-de) [reply frontend/src/contracts/paymasterVault.ts]: This resolves itself now — `TokenConfigResult` has been removed as dead code (see above). The naming difference was intentional (contract ABI naming vs domain naming), but it's moot now.

File: frontend/src/App.tsx [hunk 5]
```diff
@@ -310,14 +318,47 @@ export function App() {
...
     if (!isValidConnection) return 'Connect Wallet'
     if (paymaster.isLoading) return 'Processing...'
     if (!amount || parsedAmount === 0n) return 'Enter amount'
+    if (depositLimits.isError) return 'Unable to verify limits'
+    if (isTokenDisabled) return 'Token not supported'
+    if (isUnderMinimum) return `Below minimum (${depositLimits.formattedMin} ${selectedToken?.symbol ?? ''})`
+    if (isOverMaximum) return `Above maximum (${depositLimits.formattedMax} ${selectedToken?.symbol ?? ''})`
     if (isInsufficientBalance) return `Insufficient ${selectedToken?.symbol ?? 'Token'} Balance`
     return 'Bridge to Sapphire'
   }
```
[the remark this answers is not in the document]
Comment (rube-de) [reply frontend/src/App.tsx:365]: Good catch — the disabled-but-unlabeled state was confusing UX. Added `'Checking limits...'` after the "Enter amount" guard so it only shows when the user has typed an amount but limits haven't loaded yet. Fixed in 9c76dec.

File: frontend/src/hooks/useDepositLimits.ts [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+  maxDeposit: 0n,
+  isEnabled: false,
+  formattedMin: '0',
+  formattedMax: '0',
+  isLoading: true,
+  isError: false,
+}
+
+/**
+ * Fetches deposit limits from PaymasterVault's tokenConfig.
```
[the remark this answers is not in the document]
Comment (rube-de) [reply frontend/src/hooks/useDepositLimits.ts]: Updated to reference `getTokenConfig` (the function name) rather than `tokenConfig` (the underlying Solidity mapping). Fixed in 9c76dec.

File: frontend/src/App.tsx [hunk 7]
```diff
@@ -163,6 +164,13 @@ export function App() {
     query: { enabled: !!address, refetchInterval: 30_000 },
   })
 
+  // Deposit limits from PaymasterVault tokenConfig
```
[the remark this answers is not in the document]
Comment (rube-de) [reply frontend/src/App.tsx]: Fixed for consistency with the hook's doc comment. Both now reference `getTokenConfig`. Fixed in 9c76dec.
